src/frontend/src/instance/jaeger.rs

```diff
@@ -244,13 +261,19 @@ async fn query_trace_table(
         })?;
 
     // Apply the distinct if needed.
-    let dataframe = if distinct {
-        dataframe.distinct().context(DataFusionSnafu)?
-    } else {
-        // for non distinct query, sort by timestamp to make results stable
+    let dataframe = if !distincts.is_empty() {
         dataframe
-            .sort_by(vec![col(TIMESTAMP_COLUMN)])
+            .distinct_on(distincts.clone(), distincts, None)
             .context(DataFusionSnafu)?
+    } else {
+        dataframe
+    };
+
+    // Apply the sorts if needed.
+    let dataframe = if !sorts.is_empty() {
+        dataframe.sort_by(sorts).context(DataFusionSnafu)?
```

Comment:
`sort_by` uses the `asc` order on default. Is it intentional?

Reply:
+1